frontend/src/compliance_reporting/ScheduleCContainer.js

```diff
@@ -124,65 +124,85 @@ class ScheduleCContainer extends Component {
     }
   }
 
-  componentWillReceiveProps (nextProps, nextContext) {
-    const { grid } = this.state;
+  componentWillReceiveProps(nextProps, nextContext) {
+    const {grid} = this.state;
 
-    if (nextProps.scheduleState.scheduleC && nextProps.scheduleState.scheduleC.records) {
-      if ((grid.length - 2) < nextProps.scheduleState.scheduleC.records.length) {
-        this._addRow(nextProps.scheduleState.scheduleC.records.length - (grid.length - 2));
+    if (nextProps.snapshot) {
+      const source = nextProps.snapshot.scheduleC;
+
+      if ((grid.length - 2) < source.records.length) {
```

Comment:
Can we add a check for source here? and to make sure we can retrieve length from source.records?

I tested this with a compliance report that doesn't have schedule B filled in and it's running into errors